std/assembly/string.ts

```diff
@@ -127,6 +127,67 @@ export class String {
     );
   }
 
+  @operator("!=")
+  private static __ne(left: String, right: String): bool {
+    return !this.__eq(left, right);
+  }
+
+  @operator(">")
+  private static __gt(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = min<isize>(left.length, right.length);
+    return compare_memory(
+      changetype<usize>(left)  + HEADER_SIZE,
+      changetype<usize>(right) + HEADER_SIZE,
+      (<usize>length << 1)
+    ) > 0;
+  }
+
+  @operator(">=")
+  private static __gte(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = min<isize>(left.length, right.length);
+    return compare_memory(
+      changetype<usize>(left)  + HEADER_SIZE,
+      changetype<usize>(right) + HEADER_SIZE,
+      (<usize>length << 1)
+    ) >= 0;
+  }
+
+  @operator("<")
+  private static __lt(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = min<isize>(left.length, right.length);
+    return compare_memory(
+      changetype<usize>(left)  + HEADER_SIZE,
+      changetype<usize>(right) + HEADER_SIZE,
+      (<usize>length << 1)
+    ) < 0;
+  }
+
+  @operator("<=")
+  private static __lte(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = min<isize>(left.length, right.length);
```

Comment:
Performs two casts to isize here. Might it make sense to use min<i32> and upcast once below where <usize> is expected?

Reply:
Hmm, but `min<isize>` and `max<isize>` used everywhere. I prefer leave this changes for you if you think this makes sense.

Reply:
For [example](https://github.com/MaxGraey/assemblyscript/blob/87e034aaeddc6584e11e3e9d01e028f00ddccc1a/std/assembly/string.ts#L202)

Reply:
In your example, it is used because both arguments are already casted to isize, I believe.

Reply:
Ok, I fixed this. Hmm, but nothing changes in wast file...

Reply:
Yeah, this doesn't change anything because `isize` is just a 32-bit integer in WASM32. This is theoretical for WASM64.

Reply:
Got it